src/Model/Locale.php

```diff
@@ -329,19 +391,31 @@ public function getBaseURL()
     {
         $base = Director::baseURL();
 
-        // Build domain-specific base URL
-        if ($this->Domain() && $this->Domain()->exists()) {
-            $base = Controller::join_links(Director::protocol() . $this->Domain()->Domain, $base);
+        // Prepend hostname for domain mode
+        $domain = $this->getDomain();
+        if ($domain) {
+            $base = Controller::join_links($domain->Link(), $base);
         }
 
-        $locale = $this->getLocale();
-        // Don't append locale to home page for default locale
-        $defaultLocaleObj = $this->getDefault($this->Domain()->Domain);
-        if ($defaultLocaleObj && $locale === $defaultLocaleObj->Locale) {
-            return $base;
+        // Append locale urlsegment if a non-default locale
+        if (!$this->getIsDefault()) {
+            $base = Controller::join_links($base, $this->getURLSegment(), '/');
         }
 
-        // Append locale otherwise
-        return Controller::join_links($base, $this->getURLSegment(), '/');
+        return $base;
+    }
+
+    /**
+     * Get other locales that appear alongside this (including self)
```

Comment:
This seems strange to me, if we're calling the method "get sibling locales". I'm only commenting though, I don't mind :)